src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1288,21 +1400,31 @@ def maybe_free_model_hooks(self):
         # make sure the model is in the same state as before calling it
         self.enable_model_cpu_offload()
 
-    def enable_sequential_cpu_offload(self, gpu_id: int = 0, device: Union[torch.device, str] = "cuda"):
+    def enable_sequential_cpu_offload(self, gpu_id: Optional[int] = None, device: Union[torch.device, str] = "cuda"):
         r"""
         Offloads all models to CPU using 🤗 Accelerate, significantly reducing memory usage. When called, the state
         dicts of all `torch.nn.Module` components (except those in `self._exclude_from_cpu_offload`) are saved to CPU
         and then moved to `torch.device('meta')` and loaded to GPU only when their specific submodule has its `forward`
         method called. Offloading happens on a submodule basis. Memory savings are higher than with
         `enable_model_cpu_offload`, but performance is lower.
+
+        Arguments:
+            gpu_id (`int`, *optional*):
+                The ID of the accelerator that shall be used in inference. If not specified, it will default to 0.
+            device (`torch.Device` or `str`, *optional*, defaults to "cuda"):
+                The PyTorch device type of the accelerator that shall be used in inference. If not specified, it will
+                default to "cuda".
         """
         if is_accelerate_available() and is_accelerate_version(">=", "0.14.0"):
             from accelerate import cpu_offload
         else:
             raise ImportError("`enable_sequential_cpu_offload` requires `accelerate v0.14.0` or higher")
 
+        # _offload_gpu_id should be set to passed gpu_id or default to previously set id or default to 0
+        self._offload_gpu_id = gpu_id or self._offload_gpu_id or 0
+
         if device == "cuda":
-            device = torch.device(f"{device}:{gpu_id}")
+            device = torch.device(f"{device}:{self._offload_gpu_id }")
```

Comment:
Not sure if `f"{device}"` is robust enough for all devices. It might be!

Also, how do we deal with the user passing `device=torch.device("cuda:1")` without `gpu_id`?

Reply:
Should be handled in a nice, robust way now!